### PR TITLE
translate content/zh/docs/tasks/administer-cluster/dns-debugging-reso…

### PR DESCRIPTION
**Pull request by NickSu86**
…lution.md to chinese

>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> Please delete this note before submitting the pull request.
>
> For 1.14 Features: set Milestone to 1.14 and Base Branch to dev-1.14
> 
> For Chinese localization, base branch to release-1.12
>
> For Korean Localization: set Base Branch to dev-1.13-ko.<latest team milestone>
>
> Help editing and submitting pull requests:
> https://kubernetes.io/docs/contribute/start/#improve-existing-content.
>
> Help choosing which branch to use:
> https://kubernetes.io/docs/contribute/start#choose-which-git-branch-to-use.
>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>

### CODE REVIEW

**Comment by DanyC97** [comment]
i might be confused but should you update the specific language and not `en` ?

**Comment by NickSu86** [comment]
> i might be confused but should you update the specific language and not `en` ?

sorry, it's my mistake, I have already moved it to 'zh'

File: content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md [hunk 1]
```diff
@@ -0,0 +1,601 @@
...
+
+```shell
+kubectl get pods --namespace=kube-system -l k8s-app=kube-dns
+NAME                    READY     STATUS    RESTARTS   AGE
+...
+kube-dns-v19-ezo1y      3/3       Running   0           1h
+...
+```
+
+如果您发现没有 pod 在运行，或者这些 Pod 的状态是 failed 或者 completed ， 那可能这个 DNS 插件在您当前的环境里并没有成功部署，您将需要手动去部署它。
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md]:
```suggestion
如果您发现没有 pod 在运行，或者这些 Pod 的状态是 failed 或者 completed， 那可能这个 DNS 插件在您当前的环境里并没有成功部署，您将需要手动去部署它。
```

File: content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md [hunk 2]
```diff
@@ -0,0 +1,601 @@
...
+
+### <!--
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md]:
```suggestion
<!--
```

File: content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md [hunk 3]
```diff
@@ -0,0 +1,601 @@
...
+
+```shell
+kubectl logs --namespace=kube-system $(kubectl get pods --namespace=kube-system -l k8s-app=kube-dns -o name | head -1) -c kubedns
+
+kubectl logs --namespace=kube-system $(kubectl get pods --namespace=kube-system -l k8s-app=kube-dns -o name | head -1) -c dnsmasq
+
+kubectl logs --namespace=kube-system $(kubectl get pods --namespace=kube-system -l k8s-app=kube-dns -o name | head -1) -c sidecar
+```
+
+看日志信息里是否有可疑的错误，对于 kube-dns, 一个 '`W`', '`E`' 或者 '`F`' 开头的行表示对应的 Warning（警告）， Error（错误）或者 Failure（失败）。请搜索日志等级是否有这样的关键字的日志信息并使用 [kubernetes issues](https://github.com/kubernetes/kubernetes/issues) 来申报故障。
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md]:
```suggestion
看日志信息里是否有可疑的错误，对于 kube-dns, 一个 '`W`', '`E`' 或者 '`F`' 开头的行表示对应的 Warning（警告）， Error（错误）或者 Failure（失败）。请搜索日志等级是否有这样的关键字的日志信息并使用 [kubernetes issues](https://github.com/kubernetes/kubernetes/issues) 来提交错误报告。
```

File: content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md [hunk 4]
```diff
@@ -0,0 +1,601 @@
...
+
+```shell
+kubectl get svc --namespace=kube-system
+NAME         TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)             AGE
+...
+kube-dns     ClusterIP   10.0.0.10      <none>        53/UDP,53/TCP        1h
+...
+```
+
+注意不管是 CoreDNS 还是 kube-dns ， 这个 service 的名字都会是"kube-dns" 。
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md]:
```suggestion
注意不管是 CoreDNS 还是 kube-dns，这个 service 的名字都会是“kube-dns”。
```

File: content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md [hunk 5]
```diff
@@ -0,0 +1,601 @@
...
+如果您已经创建了这个 service 或者说在这个例子里它应该是默认自动创建的，但是它并没有出现，请阅读 [services 纠错](/docs/tasks/debug-application-cluster/debug-service/) 来获取更多信息。
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md]:
```suggestion
如果您已经创建了这个 service 或者说在这个例子里它应该是默认自动创建的，但是它并没有出现，请阅读 [services 纠错](/docs/tasks/debug-application-cluster/debug-service/)来获取更多信息。
```

File: content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md [hunk 6]
```diff
@@ -0,0 +1,601 @@
...
+If you do not see the endpoints, see endpoints section in the
+[debugging services](/docs/tasks/debug-application-cluster/debug-service/) documentation.
+
+For additional Kubernetes DNS examples, see the
+[cluster-dns examples](https://github.com/kubernetes/examples/tree/master/staging/cluster-dns)
+in the Kubernetes GitHub repository.
+
+-->
+
+### DNS 的 endpoints 暴露了吗？
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md]:
```suggestion
### DNS 端点是公开的吗？
```

File: content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md [hunk 7]
```diff
@@ -0,0 +1,601 @@
...
+
+您可以使用 `kubectl get endpoints`命令来验证 DNS 的 endpoint 是否公开了。
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md]:
```suggestion
您可以使用 `kubectl get endpoints`命令来验证 DNS 端点是否已经公开。
```

File: content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md [hunk 8]
```diff
@@ -0,0 +1,601 @@
...
+
+```shell
+kubectl get ep kube-dns --namespace=kube-system
+NAME       ENDPOINTS                       AGE
+kube-dns   10.180.3.17:53,10.180.3.17:53    1h
+```
+
+如果您没看到对应的 endpoints， 请阅读 [services 纠错](/docs/tasks/debug-application-cluster/debug-service/) 的 endpoints 小节。
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md]:
```suggestion
如果您没看到对应的 endpoints， 请阅读 [services 纠错](/docs/tasks/debug-application-cluster/debug-service/)的 endpoints 小节。
```

File: content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md [hunk 9]
```diff
@@ -0,0 +1,601 @@
...
+
+若需要更多的 Kubernetes DNS 例子，请在 Kubernetes GitHub 仓库里查看[cluster-dns examples](https://github.com/kubernetes/examples/tree/master/staging/cluster-dns) 。
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md]:
```suggestion
若需要更多的 Kubernetes DNS 例子，请在 Kubernetes GitHub 仓库里查看[cluster-dns 例子](https://github.com/kubernetes/examples/tree/master/staging/cluster-dns) 。 
```

File: content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md [hunk 10]
```diff
@@ -0,0 +1,601 @@
...
+kubectl get ep kube-dns --namespace=kube-system
+NAME       ENDPOINTS                       AGE
+kube-dns   10.180.3.17:53,10.180.3.17:53    1h
+```
+
+如果您没看到对应的 endpoints， 请阅读 [services 纠错](/docs/tasks/debug-application-cluster/debug-service/) 的 endpoints 小节。
+
+若需要更多的 Kubernetes DNS 例子，请在 Kubernetes GitHub 仓库里查看[cluster-dns examples](https://github.com/kubernetes/examples/tree/master/staging/cluster-dns) 。 
+
+### <!--
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md]:
```suggestion
<!--
```

File: content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md [hunk 11]
```diff
@@ -0,0 +1,601 @@
...
+
+```
+kubectl -n kube-system edit configmap coredns
+```
+
+-->
+
+### DNS 查询有被接收或者执行吗？
+
+您可以通过给 CoreDNS 的配置文件 (aka Corefile)添加`log`插件来判断查询是否被正确接收。
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md]:
```suggestion
您可以通过给 CoreDNS 的配置文件（也叫 Corefile）添加 `log` 插件来判断查询是否被正确接收。
```

File: content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md [hunk 12]
```diff
@@ -0,0 +1,601 @@
...
+2018/08/15 14:37:15 [INFO] linux/amd64, go1.10.3, 2e322f6
+CoreDNS-1.2.0
+linux/amd64, go1.10.3, 2e322f6
+2018/09/07 15:29:04 [INFO] plugin/reload: Running configuration MD5 = 162475cdf272d8aa601e6fe67a6ad42f
+2018/09/07 15:29:04 [INFO] Reloading complete
+172.17.0.18:41675 - [07/Sep/2018:15:29:11 +0000] 59925 "A IN kubernetes.default.svc.cluster.local. udp 54 false 512" NOERROR qr,aa,rd,ra 106 0.000066649s
+
+```
+
+## <!--
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md]:
```suggestion
<!--
```

File: content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md [hunk 13]
```diff
@@ -0,0 +1,601 @@
...
+loop when resolving names in upstream servers. This can be fixed manually by using kubelet's `--resolv-conf` flag
+to point to the correct `resolv.conf` (With `systemd-resolved`, this is `/run/systemd/resolve/resolv.conf`).
+kubeadm (>= 1.11) automatically detects `systemd-resolved`, and adjusts the kubelet flags accordingly.
+
+-->
+
+## 已知问题
+
+有些 Linux 发行版本 (比如 Ubuntu)， 默认使用一个本地的 DNS 解析器 (systemd-resolved)。
+Systemd-resolved 会用一个 stub file 来覆盖 `/etc/resolv.conf`从而在解析域名的时候导致了重复向 DNS 上游服务器推送请求。 这个问题可以通过手动指定 kubelet 的 `--resolv-conf` 标签为正确的 `resolv.conf` (如果是 `systemd-resolved`，则这个文件路径为 `/run/systemd/resolve/resolv.conf`) 来解决。
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md]:
```suggestion
Systemd-resolved 会用一个 stub 文件来覆盖 `/etc/resolv.conf`从而在解析域名的时候导致了重复向 DNS 上游服务器推送请求。 这个问题可以通过手动指定 kubelet 的 `--resolv-conf` 标签为正确的 `resolv.conf` (如果是 `systemd-resolved`，则这个文件路径为 `/run/systemd/resolve/resolv.conf`) 来解决。
```

File: content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md [hunk 14]
```diff
@@ -0,0 +1,601 @@
...
+If you are using Alpine version 3.3 or earlier as your base image, DNS may not
+work properly owing to a known issue with Alpine.
+Check [here](https://github.com/kubernetes/kubernetes/issues/30215)
+for more information.
+
+-->
+
+Kubernetes 的安装并不会默认配置节点的 `resolv.conf` 文件来使用集群的 DNS 服务，因为这个配置对于不同的发行版本是不一样的。这个问题应该迟早会被解决的。
+
+Linux 的 libc 会在仅有三个 DNS 的 `nameserver` 和六个 DNS 的`search` 记录时会不可思议的卡死 ([详情请查阅这个2005年的bug](https://bugzilla.redhat.com/show_bug.cgi?id=168253))。Kubernetes 需要占用一个 `nameserver` 记录和三个`search`记录。这意味着如果一个本地的安装已经使用了三个`nameserver`或者使用了超过三个的 `search`记录，那有些配置很可能会丢失。有一个不完整的解决方案就是在节点上使用`dnsmasq`来提供更多的`nameserver`配置，但是无法提供更多的`search`记录。您也可以使用kubelet 的 `--resolv-conf` 标签来解决这个问题。
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md]:
```suggestion
Linux 的 libc 会在仅有三个 DNS 的 `nameserver` 和六个 DNS 的 `search` 记录时会不可思议的卡死 ([详情请查阅这个2005年的 bug](https://bugzilla.redhat.com/show_bug.cgi?id=168253))。Kubernetes 需要占用一个 `nameserver` 记录和三个 `search` 记录。这意味着如果一个本地的安装已经使用了三个 `nameserver` 或者使用了超过三个的 `search` 记录，那有些配置很可能会丢失。有一个不完整的解决方案就是在节点上使用 `dnsmasq` 来提供更多的 `nameserver` 配置，但是无法提供更多的 `search` 记录。您也可以使用 kubelet 的 `--resolv-conf` 标签来解决这个问题。
```

File: content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md [hunk 15]
```diff
@@ -0,0 +1,601 @@
...
+
+如果您是使用 Alpine  3.3 或者更早版本作为您的基础镜像， DNS 可能会由于Alpine 一个已知的问题导致无法正常工作，请查看[这里](https://github.com/kubernetes/kubernetes/issues/30215)获取更多资料。
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md]:
```suggestion
如果您是使用 Alpine 3.3 或者更早版本作为您的基础镜像，DNS 可能会由于 Alpine 一个已知的问题导致无法正常工作，请查看[这里](https://github.com/kubernetes/kubernetes/issues/30215)获取更多资料。
```

File: content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md [hunk 16]
```diff
@@ -0,0 +1,601 @@
...
+way the Kubernetes cluster DNS server processes DNS queries, to facilitate
+the lookup of federated services (which span multiple Kubernetes clusters).
+See the [Cluster Federation Administrators' Guide](/docs/concepts/cluster-administration/federation/)
+for more details on Cluster Federation and multi-site support.
+
+## -->
+
+## Kubernetes Federation (支持多区域部署)
+
+自从1.3 版本支持了多个 Kubernetes 的联邦集群后，集群 DNS 服务在处理 DNS 请求时需要有一些微弱的调整 (这是向下兼容的)，从而可以使用跨越多个 Kubernetes 集群的联邦服务。请看 [联邦集群管理向导](/docs/concepts/cluster-administration/federation/) 获取更多关于联邦集群和多点支持的信息。
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md]:
```suggestion
自从 1.3 版本支持了多个 Kubernetes 的联邦集群后，集群 DNS 服务在处理 DNS 请求时需要有一些微弱的调整 (这是向下兼容的)，从而可以使用跨越多个 Kubernetes 集群的联邦服务。请看 [联邦集群管理向导](/docs/concepts/cluster-administration/federation/) 获取更多关于联邦集群和多点支持的信息。
```

File: content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md [hunk 17]
```diff
@@ -0,0 +1,601 @@
...
+## -->
+
+## 参考
+
+- [Services 和 Pods 的 DNS 指南](/docs/concepts/services-networking/dns-pod-service/)
+- [kube-dns DNS 插件文档](http://releases.k8s.io/{{< param "githubbranch" >}}/cluster/addons/dns/kube-dns/README.md)
+
+## 接下来
+
+- [集群里自动伸缩 DNS Service ](/docs/tasks/administer-cluster/dns-horizontal-autoscaling/).
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/dns-debugging-resolution.md]:
```suggestion
- [集群里自动伸缩 DNS Service](/docs/tasks/administer-cluster/dns-horizontal-autoscaling/).
```

**Comment by NickSu86** [comment]
> 翻译的挺好的，提了一些建议。

谢谢，我根据您的建议修改了

**Pull request merged**